---
description: コンパイラの警告 (レベル 1) CS1058
title: コンパイラの警告 (レベル 1) CS1058
ms.date: 07/20/2015
f1_keywords:
- CS1058
helpviewer_keywords:
- CS1058
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
ms.openlocfilehash: 77097472f95d636a6ebc79a5a86083042a48ca32
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: HT
ms.contentlocale: ja-JP
ms.lasthandoff: 08/30/2020
ms.locfileid: "89139126"
---
# <a name="compiler-warning-level-1-cs1058"></a><span data-ttu-id="6856d-103">コンパイラの警告 (レベル 1) CS1058</span><span class="sxs-lookup"><span data-stu-id="6856d-103">Compiler Warning (level 1) CS1058</span></span>
<span data-ttu-id="6856d-104">前の catch 句は、すべての例外を既にキャッチしています。</span><span class="sxs-lookup"><span data-stu-id="6856d-104">A previous catch clause already catches all exceptions.</span></span> <span data-ttu-id="6856d-105">スローされる例外はすべて System.Runtime.CompilerServices.RuntimeWrappedException にラップされます。</span><span class="sxs-lookup"><span data-stu-id="6856d-105">All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException</span></span>  
  
 <span data-ttu-id="6856d-106">`catch (System.Exception e)` ブロックの後の `catch()` ブロックで例外の型が指定されていない場合、この属性では CS1058 が発生します。</span><span class="sxs-lookup"><span data-stu-id="6856d-106">This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block.</span></span> <span data-ttu-id="6856d-107">警告では、`catch()` ブロックがどのような例外もキャッチしないことが示されます。</span><span class="sxs-lookup"><span data-stu-id="6856d-107">The warning advises that the `catch()` block will not catch any exceptions.</span></span>  
  
<span data-ttu-id="6856d-108">AssemblyInfo.cs ファイルで `RuntimeCompatibilityAttribute` が false に設定されている場合 (`[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`)、`catch (System.Exception e)` ブロックの後の `catch()` ブロックは CLS 以外の例外をキャッチできます。</span><span class="sxs-lookup"><span data-stu-id="6856d-108">A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span></span> <span data-ttu-id="6856d-109">この属性が明示的に false に設定されていない場合、スローされた非 CLS 例外はすべて Exceptions としてラップされ、`catch (System.Exception e)` ブロックでキャッチされます。</span><span class="sxs-lookup"><span data-stu-id="6856d-109">If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them.</span></span> <span data-ttu-id="6856d-110">詳細については、「[CLS 準拠でない例外をキャッチする方法](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md)」を参照してください。</span><span class="sxs-lookup"><span data-stu-id="6856d-110">For more information, see [How to catch a non-CLS exception](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span></span>
  
## <a name="example"></a><span data-ttu-id="6856d-111">例</span><span class="sxs-lookup"><span data-stu-id="6856d-111">Example</span></span>  
 <span data-ttu-id="6856d-112">次の例では CS1058 が生成されます。</span><span class="sxs-lookup"><span data-stu-id="6856d-112">The following example generates CS1058.</span></span>  
  
```csharp  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass
{  
   static void Main()
   {  
      try {}  
  
      catch (System.Exception e) {
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```
